packages/elements/src/clock/index.ts

```diff
@@ -482,28 +507,68 @@ export class Clock extends ResponsiveElement {
   * @param event Event Object
   * @returns {void}
   */
-  private handleUpKey (event: KeyboardEvent): void {
-    this.shift(UP, this.getShiftAmountFromTarget(event.target));
+  private handleUpKey (): void {
+    const selection = this.renderRoot.querySelector(`[part~=${this.active}]`);
+    this.shift(UP, this.getShiftAmountFromTarget(selection));
+    this.updateAriaValueText();
   }
 
   /**
   * Handle DOWN key press
   * @param event Event Object
   * @returns {void}
   */
-  private handleDownKey (event: KeyboardEvent): void {
-    this.shift(DOWN, this.getShiftAmountFromTarget(event.target));
+  private handleDownKey (): void {
+    const selection = this.renderRoot.querySelector(`[part~=${this.active}]`);
+    this.shift(DOWN, this.getShiftAmountFromTarget(selection));
+    this.updateAriaValueText();
+  }
+
+  /**
+  * Handles Left key press
+  * @param event Event Object
+  * @returns {void}
+  */
+  private handleLeftRightKey (): void {
+    const segment = this.active === 'hours' ? 'minutes' : 'hours';
+    this.changeActiveSegment(segment);
+  }
+
+  /**
+  * Updates active attribute for increment and decrement button
+  * by new segment
+  * @param segment new active segment
+  * @returns {void}
+  */
+  private changeActiveSegment (segment: string) {
+    this.renderRoot.querySelectorAll('[active]')?.forEach((el) => el.removeAttribute('active'));
+    this.active = segment;
+    this.renderRoot.querySelector(`[part~="${this.active}"]`)?.querySelector('[part="increment-button"]')?.setAttribute('active', '');
+    this.renderRoot.querySelector(`[part~="${this.active}"]`)?.querySelector('[part="decrement-button"]')?.setAttribute('active', '');
+  }
+
+  /**
+  * Updates aria-value-text to have the same value as screen
+  * @returns {void}
+  */
+  private updateAriaValueText () {
+    const value = `Time: ${padNumber(this.displayHours, 2)}:${padNumber(this.displayMinutes, 2)}${this.showSeconds ? ':' + padNumber(this.displaySeconds, 2) : ''}${this.amPm ? ' ' + this.displayAmPm : ''}`;
+    this.ariaValueText = value;
```

Comment:
Do not use AOM model, it is still not supported by Firefox and not yet a standard